platform/commonUI/browse/src/ObjectHeaderController.js

```diff
@@ -41,25 +41,44 @@ define(
          * @param event the mouse event
          */
         ObjectHeaderController.prototype.updateName = function (event) {
-            if (event && (event.type === 'blur' || event.which === 13)) {
-                var name = event.currentTarget.innerHTML;
+            if (!event) {
+                return;
+            }
+
+            if (this.enterKeyHandled === true) {
+                this.enterKeyHandled = false;
+                return;
+            }
+
+            if (event.type === 'blur') {
+                this.updateModel(event);
+            } else if (event.which === 13) {
+                this.updateModel(event);
+                this.enterKeyHandled = true;
+                event.currentTarget.blur();
```

Comment:
Locally, I'm still seeing the text input keep focus (and add newlines) when I hit enter.

Reply:
This resolved locally when I added a `window.getSelection().removeAllRanges();` after the `blur()`

Reply:
Will add that line after the blur().